lib/oli_web/live/community_live/show_view.ex

```diff
@@ -338,31 +343,50 @@ defmodule OliWeb.CommunityLive.ShowView do
     {:noreply, socket}
   end
 
-  def handle_event("add_institution", %{"name" => name}, socket) do
+  def handle_event(
+        "add_institution",
+        %{"institution_id" => ""},
+        socket
+      ) do
+    socket = clear_flash(socket)
+    socket = put_flash(socket, :error, "Please select an institution.")
+    {:noreply, socket}
+  end
+
+  def handle_event(
+        "add_institution",
+        %{"institution_id" => institution_id},
+        socket
+      ) do
     socket = clear_flash(socket)
     community_id = socket.assigns.community.id
+    institution_id = String.to_integer(institution_id)
 
-    names =
-      name
-      |> String.split(",")
-      |> Enum.map(&String.trim(&1))
+    institutions_added = socket.assigns.community_institutions |> Enum.map(& &1.id)
 
-    socket =
-      case Groups.create_community_institutions_from_names(names, %{
-             community_id: community_id
-           }) do
-        {:ok, _community_institutions} ->
-          put_flash(socket, :info, "Community institution(s) successfully added.")
+    if institution_id in institutions_added do
+      socket = put_flash(socket, :error, "Institution has already been added to the community.")
+      {:noreply, socket}
+    else
+      socket =
+        case Groups.create_community_institutions_from_ids(institution_id |> List.wrap(), %{
```

Comment:
Could the `List.wrap` be substituted by adding the `[]` directly to the `institution_id`? Also, the argument passed to the `case` is a little bit longer. What do you think binding it to a variable?

Reply:
Okay, I guess I'd better change it to `[institution_id]`.